src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        int hash = hash(key);
+        int index = index(hash, table.length);
 
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (current.hash == hash && Objects.equals(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
+        Node<K, V> newNode = new Node<>(hash, key, value, table[index]);
+        table[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = index(hash, table.length);
+
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (current.hash == hash && Objects.equals(current.key, key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void resize() {
+        Node<K, V>[] oldTable = table;
+        int newCapacity = oldTable.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+
+        threshold = (int) (newCapacity * LOAD_FACTOR);
+        table = newTable;
+        size = 0;
+
+        for (Node<K, V> oldNode : oldTable) {
+            while (oldNode != null) {
+                put(oldNode.key, oldNode.value);
+                oldNode = oldNode.next;
+            }
+        }
+    }
```

Comment:
The `resize` method should not be public as it is only used internally within the `MyHashMap` class. Change its access modifier to private to adhere to the encapsulation principle.